src/hooks/useActiveDescendantListFocus/useActiveDescendantListFocusHooks.ts

```diff
@@ -134,8 +152,16 @@ export function useSupportPressItemKeyboardNavigation({
   });
 }
 
-export function useCleanVisualFocusOnBlur({ focusedElementRef, visualFocusItemIndex, setVisualFocusItemIndex }) {
-  const previousFocusedElementRef = usePrevious(focusedElementRef);
+export function useCleanVisualFocusOnBlur({
+  focusedElementRef,
+  visualFocusItemIndex,
+  setVisualFocusItemIndex
+}: {
+  focusedElementRef: MutableRefObject<HTMLElement>;
+  visualFocusItemIndex: number;
+  setVisualFocusItemIndex: (index: number) => unknown;
+}) {
+  const previousFocusedElementRef = usePrevious<MutableRefObject<HTMLElement>>(focusedElementRef);
```

Comment:
I'm curious - can't `usePrevious` detect the type of `focusedElementRef` automatically? or do you have to specify the types for the generics to work?

Reply:
it didn't 🤷 , maybe i didn't write the generic well enough @laviomri